aws/modules/vpc/main.tf

```
@@ -0,0 +1,71 @@
+data "aws_availability_zones" "available_zones" {
+  state = "available"
+}
+
+resource "aws_vpc" "business_service" {
+  cidr_block           = var.cidr_block
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+  tags                 = var.tags
+}
+
+resource "aws_subnet" "public" {
+  count                   = length(var.public_subnets) > 0 ? length(var.public_subnets) : 0
+  cidr_block              = element(var.public_subnets, count.index)
+  availability_zone       = data.aws_availability_zones.available_zones.names[count.index]
+  vpc_id                  = aws_vpc.business_service.id
+  map_public_ip_on_launch = true
+  tags                    = var.tags
+}
+
+resource "aws_subnet" "private" {
+  count             = length(var.private_subnets) > 0 ? length(var.private_subnets) : 0
+  cidr_block        = element(var.private_subnets, count.index)
+  availability_zone = data.aws_availability_zones.available_zones.names[count.index]
+  vpc_id            = aws_vpc.business_service.id
+  tags              = var.tags
+}
+
+resource "aws_internet_gateway" "gateway" {
```

if we don't have any public subnet we don't have to create a gateway
```suggestion
resource "aws_internet_gateway" "gateway" {
  count                  = length(var.public_subnets) > 0 ? 1 : 0
```

---

The `gateway` is also being used by the `nat` for the private subnets. So we should keep it unless we are planning to remove also the `nat` for the private subnets.